### ISSUE TITLE
Add `BundleRef` members to `AnonymousBundles`

### ISSUE THREAD

**Issue by dan-fritchman** [dan-fritchman/Hdl21#22]
Minimal test case: 

```python

def test_anon_bundle_refs():
    """ Test adding `BundleRef`s to `AnonymousBundle`s. """
    import hdl21 as h 

    @h.bundle
    class Diff:  # Our favorite Bundle: the differential pair
        p, n = h.Signals(2)

    @h.module
    class HasDiff:  # Module with a `Diff` Port
        d = Diff(port=True)

    @h.module
    class HasHasDiff:  # Module instantiating a few `HasDiff`
        d = Diff()
        # Instance connected to the Bundle Instance
        h1 = HasDiff(d=d)

        # THE POINT OF THE TEST: 
        # Instance with an Anon Bundle, "equal to" the Bundle Instance
        h2 = HasDiff(d=h.AnonymousBundle(p=d.p, n=d.n))
        # Instance with (p, n) flipped 
        h3 = HasDiff(d=h.AnonymousBundle(p=d.n, n=d.p))

    h.elaborate(HasHasDiff)
```

This is a very common and helpful thing to want to do. (In fact it was very much this use-case that led me to the problem.) But, it'll need a few additions to support. 

Currently fails with: 

```
E           TypeError: Invalid Bundle attribute for `<hdl21.bundle.AnonymousBundle object at 0x7f40ef9cad90>`: `<hdl21.bundle.BundleRef object at 0x7f40ef9cafa0>`
```

Should require: 

* Add a place to store the `BundleRef`s on `AnonymousBundle`. There is an unused field `refs` which I believe was intended for this. 
* Add elaboration handling to resolve the `BundleRef`s to `Signal`s. The `BundleFlattener` pass appears the natural place, and does something similar for the existing `Signal` and `BundleInstance` members of each `AnonymousBundle`.

**Comment by dan-fritchman** [comment]
Commits starting with [e55ace8](https://github.com/dan-fritchman/Hdl21/commit/e55ace89682a458dbae0bdc807ee67eae444a9aa) include this as a (failing) test.

**Comment by dan-fritchman** [comment]
This will be resolved by PR #36.
